### ISSUE TITLE
[a11y]: Contained List [with action & interactive items / actions] - VO doesn't announce the label dismiss button in iPhone

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13939]
### Package

@carbon/react

### Browser

Iphone Safari

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset

Manual QA

### Assistive technology

Voice Over

### Description

Voiceover iPhone Safari does not announce the label for the dismiss button

### User impact
When the screen reader is unable to identify the label, it’s difficult for users to understand the purpose of the control.

### WCAG 2.1 Violation
1.3.1 Info and Relationships 

### Reproduction/example

Record `Number 27` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

https://react.carbondesignsystem.com/?path=/story/components-containedlist--with-interactive-items-and-actions

https://github.com/carbon-design-system/carbon/assets/32720851/a54b0e9b-fc70-4c12-8660-6ec8576ba73d

### Steps to reproduce

Voiceover iPhone Safari does not announce the label for the dismiss button. however, . The label is announced for the last control only when The user swipes back word from the end of the screen. And announced the label for the first control only When user swipes backward from the beginning of the screen. the label is never announced for the controls present in between.

1. open this link in iPhone Safari: https://react.carbondesignsystem.com/iframe.html?args=&id=components-containedlist--with-interactive-items-and-actions&viewMode=story

2. Go to Setting > Accessibility > VoiceOver on 

2. Take 2 fingers and swipe down to hear VO navigate through the page

Note: Working fine in jaws and in Android talkback

### Recommended fix
Screen reader is expected to announce the label for all the controls consistently

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
I think this may be an issue with Button / Tooltip
